app/services/data/get-claim-list-for-advanced-search.js

```diff
@@ -0,0 +1,110 @@
+const config = require('../../../knexfile').intweb
+const knex = require('knex')(config)
+const moment = require('moment')
+
+var countQuery
+var selectQuery
+
+var validSearchOptions = [
+  'reference',
+  'name',
+  'ninumber',
+  'prisonerNumber',
+  'prison'
+]
+
+module.exports = function (searchCriteria, offset, limit) {
+  var validSearchOptionFound = false
+
+  for (var option in searchCriteria) {
+    if (validSearchOptions.indexOf(option) !== -1) {
+      if (searchCriteria[option]) {
+        validSearchOptionFound = true
+      }
+    }
+  }
+
+  if (!validSearchOptionFound) {
+    return Promise.resolve({
+      claims: [],
+      total: 0
+    })
+  }
+
+  createBaseQueries(limit, offset)
+
+  if (searchCriteria.reference) {
+    applyReferenceFilter(countQuery, searchCriteria.reference)
+    applyReferenceFilter(selectQuery, searchCriteria.reference)
+  }
+
+  if (searchCriteria.name) {
+    applyNameFilter(countQuery, searchCriteria.name)
+    applyNameFilter(selectQuery, searchCriteria.name)
+  }
+
+  if (searchCriteria.ninumber) {
+    applyNINumberFilter(countQuery, searchCriteria.ninumber)
+    applyNINumberFilter(selectQuery, searchCriteria.ninumber)
+  }
+
+  if (searchCriteria.prisonerNumber) {
+    applyPrisonerNumberFilter(countQuery, searchCriteria.prisonerNumber)
+    applyPrisonerNumberFilter(selectQuery, searchCriteria.prisonerNumber)
+  }
+
+  if (searchCriteria.prison) {
+    applyPrisonFilter(countQuery, searchCriteria.prison)
+    applyPrisonFilter(selectQuery, searchCriteria.prison)
+  }
+
+  return countQuery
+    .then(function (count) {
+      return selectQuery
+        .then(function (claims) {
+          claims.forEach(function (claim) {
+            claim.DateSubmittedFormatted = moment(claim.DateSubmitted).format('DD/MM/YYYY - HH:mm')
+            claim.Name = claim.FirstName + ' ' + claim.LastName
+          })
+          return {
+            claims: claims,
+            total: count[0]
+          }
+        })
+    })
+
+  function applyReferenceFilter (query, reference) {
+    query.where('Claim.Reference', 'like', `%${reference}%`)
+  }
+
+  function applyNameFilter (query, name) {
+    query.whereRaw(`CONCAT(Visitor.FirstName, ' ', Visitor.LastName) like '%${name}%'`)
```

Comment:
This line is potentially open to a SQL injection attack.  Refactor so that a parameter binding is used.  See http://knexjs.org/#Builder-whereRaw or give me a shout